File: clang/test/Modules/redefinition-same-header.m
// RUN: rm -rf %t.tmp
// RUN: %clang_cc1 -fsyntax-only -I %S/Inputs/SameHeader -fmodules \
// RUN:   -fimplicit-module-maps -fmodules-cache-path=%t.tmp %s -verify

// expected-error@Inputs/SameHeader/C.h:3 {{redefinition of 'c'}}
// expected-note-re@Inputs/SameHeader/B.h:3 {{'{{.*}}/C.h' included multiple times, additional (likely non-modular) include site in module 'X.B'}}
// expected-note-re@Inputs/SameHeader/module.modulemap:6 {{consider adding '{{.*}}/C.h' as part of 'X.B' definition in}}

#include "A.h" // maps to a modular
#include "C.h" // textual include

File: clang/test/Sema/redefinition-same-header.c
// RUN: rm -rf %t
// RUN: mkdir -p %t
// RUN: echo 'int yyy = 42;' > %t/a.h
// RUN: %clang_cc1 -fsyntax-only %s -I%t  -verify

// expected-error@a.h:1 {{redefinition of 'yyy'}}
// expected-note-re@redefinition-same-header.c:9 {{'{{.*}}/a.h' included multiple times, consider augmenting this header with #ifdef guards}}

#include "a.h"
#include "a.h"

int foo() { return yyy; }
